components/resolved_ts/src/endpoint.rs

```diff
@@ -78,55 +80,80 @@ impl ObserveRegion {
 
     fn track_change_log(&mut self, change_logs: &[ChangeLog]) -> Result<()> {
         match self.resolver_status {
-            ResolverStatus::Pending { ref mut locks, .. } => {
+            ResolverStatus::Pending {
+                ref mut locks,
+                ref mut tracked_index,
+                ..
+            } => {
                 for log in change_logs {
                     match log {
-                        ChangeLog::Error(e) => return Err(Error::request(e.clone())),
-                        ChangeLog::Rows { rows, .. } => rows.iter().for_each(|row| match row {
-                            ChangeRow::Prewrite { key, start_ts, .. } => {
-                                locks.push(PendingLock::Track {
+                        ChangeLog::Error(e) => {
```

Comment:
Why not return?

Reply:
Return an error will require to re-register the region, and I think it is unnecessary since an error (like `EpochNotMatch`) won't change the lock data that already write to kv engine at the apply thread. /cc @5kbpers

Reply:
After region merge some locks could be merged, so better to return an error here or detect the type of region error.